### PR TITLE
fix: fallback path include unused fills on a used source

### PR DESCRIPTION
**Pull request by dekz**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/test/market_operation_utils_test.ts [hunk 1]
```diff
@@ -1166,28 +1166,6 @@ describe('MarketOperationUtils tests', () => {
                 expect(orderSources.sort()).to.deep.eq(expectedSources.sort());
             });
 
-            it('fallback orders use different sources', async () => {
```
Comment (dekz) [inline packages/asset-swapper/test/market_operation_utils_test.ts:1169]: We don't use a different source. We can pad more to the currently used source or add a different source. Tests in `path_test.ts`

File: packages/asset-swapper/src/utils/market_operation_utils/index.ts [hunk 2]
```diff
@@ -727,6 +705,44 @@ export class MarketOperationUtils {
             }),
         );
     }
+
+    // tslint:disable-next-line: prefer-function-over-method
```
Comment (kimpers) [inline packages/asset-swapper/src/utils/market_operation_utils/index.ts:709]: I vote we just get rid of this annoying linting rule

**Review by kimpers — COMMENTED** [review]
Post merge approval ✅  😂

File: packages/asset-swapper/src/utils/market_operation_utils/path.ts [hunk 3]
```diff
@@ -99,14 +99,18 @@ export class Path {
...
+
         this.fills = [
             // Append all of the native fills first
             ...nativeFills.filter(f => f !== lastNativeFillIfExists),
             // Add the other fills that are not native in the optimal path
             ...otherFills,
-            // Add the fallbacks to the end that aren't already included
-            ...fallback.fills.filter(f => !otherSourcePathIds.includes(f.sourcePathId)),
+            // Add the fills to the end that aren't already included
+            ...fallback.fills.filter(f => !otherFillIds.includes(fillToFillId(f))),
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/utils/market_operation_utils/path.ts:113]: Can this cause non-contiguous sources to occur? Like if the current path is `[Native, Uniswap(0)]` and we create a fallback of `[Balancer, Uniswap(0), Uniswap(1)]`, do we wind up with `[Native, Uniswap(0), Balancer, Uniswap(1)]`?

**Pull request merged**